superset/datasets/columns/api.py

```diff
@@ -36,14 +38,99 @@
     method_permission_name = MODEL_API_RW_METHOD_PERMISSION_MAP
-
+    list_columns = [  # See DatasetRestApi.show_select_columns.
+        "advanced_data_type",
+        "changed_on",
+        "column_name",
+        "created_on",
+        "description",
+        "expression",
+        "filterable",
+        "groupby",
+        "id",
+        "is_active",
+        "extra",
+        "is_dttm",
+        "python_date_format",
+        "type",
+        "uuid",
+        "verbose_name",
+    ]
+    search_columns = ["table_id"]
+    max_page_size = -1
+    openapi_spec_tag = "Datasets"
     resource_name = "dataset"
-    allow_browser_login = True
 
-    openapi_spec_tag = "Datasets"
+    @expose("/<int:pk>/column", methods=["GET"])
+    @protect()
+    @safe
+    @permission_name("get")
+    @rison(get_list_schema)
+    def get_list(  # pylint: disable=arguments-differ
+        self,
+        pk: int,
+        **kwargs: Any,
+    ) -> Response:
+        """Get a list of dataset columns
+        ---
+        get:
+          description: >-
+            Get a list of dataset columns
+          parameters:
+          - in: path
+            schema:
+              type: integer
+            description: The dataset id for these columns
+            name: pk
+          - in: query
+            name: q
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/get_list_schema'
+          responses:
+            200:
+              description: Columns from dataset
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      ids:
+                        description: >-
+                          A list of column ids
+                        type: array
+                        items:
+                          type: string
+                      count:
+                        description: >-
+                          The total record count on the backend
+                        type: number
+                      result:
+                        description: >-
+                          The result from the get list query
+                        type: array
+                        items:
+                          $ref: '#/components/schemas/{{self.__class__.__name__}}.get_list'  # pylint: disable=line-too-long
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+
+        rison_data = kwargs.setdefault("rison", {})
+        rison_data.setdefault("filters", [])
+        rison_data["filters"].append({"col": "table_id", "opr": "eq", "value": pk})
+        rison_data["page_size"] = -1
+        return self.get_list_headless(**kwargs)
```

Comment:
@dpgaspar, @michael-s-molina, or @villebro I would love your input on this as I'm not overly familiar with the new FAB API and/or how best to define said endpoint. I came across the pattern by prodding around.

Reply:
@dpgaspar I added the `rison_data["page_size"] = -1` as the `/api/v1/dataset/{pk}` response is non-paginated and I wanted to ensure consistency. Additionally this endpoint is currently solely used in a non-paginated way, i.e., it's not powering any `/*/list/` endpoints.

Reply:
Regarding the pattern I see it being only used twice and it only applies one rison override (as opposed to two which is needed in our case). I think there's merit in refactoring if it's used more than once. Furthermore [this](https://github.com/apache/superset/blob/d3274374625b44c3f4c5de8b30fd44b428e1abf3/superset/reports/logs/api.py#L84-L92) example is actually a misnomer in terms of naming, i.e., the function name has been copied and pasted as is even though it's applying the `report_schedule` as opposed to `layer`.

Reply:
I Agree let's refactor this after. Still think we need to apply the `DatasourceFilter` here, have you checked it?